clarity/src/vm/ast/traits_resolver/mod.rs

```diff
@@ -63,7 +64,9 @@ impl TraitsResolver {
                     match (&args[0].pre_expr, &args[1].pre_expr) {
                         (Atom(trait_name), List(trait_definition)) => {
                             // Check for collisions
-                            if contract_ast.referenced_traits.contains_key(trait_name) {
+                            if let Entry::Occupied(_) =
```

Comment:
Using `Entry` will help performance if it's replacing both a check and an insert, because it avoids needing to hash the key twice. If you aren't using it for the insert, you might as well just use `contains_key()`. In order to make sense here, you need use the vacant entry later for the insert a few lines later